result

ADIGrimm opened the pull request.

ADIGrimm committed:
result

ADIGrimm committed:
checkstyle fix

ADIGrimm committed:
checkstyle fix?

Elena-Bruyako commented on src/main/java/online/bookstore/OnlineBookstoreApplication.java:
```diff
@@ -0,0 +1,39 @@
+import org.springframework.context.annotation.Bean;
+
+@SpringBootApplication
+@EntityScan(basePackages = "online.bookstore.model")
```
remove

Elena-Bruyako commented on src/main/java/online/bookstore/OnlineBookstoreApplication.java:
```diff
@@ -0,0 +1,39 @@
+                    + "controlled by technology and the government.");
+            book.setCoverImage("https://m.media-amazon.com/images/I/81gTwYAhU7L.jpg");
+            book.setIsbn("9780060850524");
+            bookService.save(book);
```
call also findAll method

Elena-Bruyako commented on src/main/java/online/bookstore/model/Book.java:
```diff
@@ -0,0 +1,83 @@
+import java.math.BigDecimal;
+import org.antlr.v4.runtime.misc.NotNull;
+
+@Entity
```
```suggestion
@Entity
@Getter
@Setter
@Table(name...
```

Elena-Bruyako commented on src/main/java/online/bookstore/model/Book.java:
```diff
@@ -0,0 +1,83 @@
+    private String description;
+    private String coverImage;
+
+    public Long getId() {
```
remove gettes and setters, you can use Lombok annotations for them

Elena-Bruyako commented on src/main/java/online/bookstore/repository/impl/BookRepositoryImpl.java:
```diff
@@ -0,0 +1,53 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
```
```suggestion
@Repository
@RequiredArgsConstructor
```

Elena-Bruyako commented on src/main/java/online/bookstore/repository/impl/BookRepositoryImpl.java:
```diff
@@ -0,0 +1,53 @@
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
```
```suggestion
```

Elena-Bruyako commented on src/main/java/online/bookstore/repository/impl/BookRepositoryImpl.java:
```diff
@@ -0,0 +1,53 @@
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't add a book to db: " + book, e);
```
```suggestion
            throw new DataProcessingException("Can't add a book to db: " + book, e);
```

Elena-Bruyako commented on src/main/java/online/bookstore/repository/impl/BookRepositoryImpl.java:
```diff
@@ -0,0 +1,53 @@
+            Query<Book> getAllBooksQuery = session.createQuery("FROM Book", Book.class);
+            return getAllBooksQuery.getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Can't get list of all books" + e);
```
```suggestion
            throw new DataProcessingException("Can't get list of all books" + e);
```

Elena-Bruyako commented on src/main/java/online/bookstore/service/impl/BookServiceImpl.java:
```diff
@@ -0,0 +1,28 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
```
```suggestion
@Service
@RequiredArgsConstructor
```

Elena-Bruyako commented on src/main/java/online/bookstore/service/impl/BookServiceImpl.java:
```diff
@@ -0,0 +1,28 @@
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+
+    @Autowired
+    public BookServiceImpl(BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
+    }
```
```suggestion
```

Elena-Bruyako reviewed (changes requested):
see comments

ADIGrimm committed:
Remove @EntityScan from OnlineBookstoreApplication
Call findAll() in OnlineBookstoreApplication
Replaced getters/setters with @Getter/@Setter in Book
Add @RequiredArgsConstructor to Book repository and service
Create DataProcessingException
